types/cypress-image-snapshot/plugin.d.ts

```diff
@@ -1,5 +1,28 @@
-/// <reference types="Cypress" />
+/* Definitions extracted from Cypress 4.5, which is the minimum
+ * version required by the cypress-image-snapshot.
+ * https://github.com/cypress-io/cypress/blob/v4.5.0/cli/types/index.d.ts#L4513-L4524
+ *
+ * The definitions have been copied in, as DefinitelyTyped does not (yet) support
+ * peerDependencies. This is causing with npms version resolution for consumers.
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20290#issuecomment-589738371
+ * this has been addressed before on these definitions here
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41921
+ *
+ * The Cypress interfaces have been stripped down, according to consumed fields
+ * in https://github.com/jaredpalmer/cypress-image-snapshot/blob/master/src/plugin.js
+ */
+type Task = (value: any) => any;
 
-declare function addMatchImageSnapshotPlugin(on: Cypress.PluginEvents, config: Cypress.PluginConfigOptions): void;
+interface Tasks {
+    [key: string]: Task;
+}
+
+interface PluginEvents {
```

Comment:
I don't work much in this repository but would it be worth exporting this interface and using it in our test file so it doesn't have to be duplicated?

Reply:
i was thinking the same. but unfortunately am also no expert in DefinitelyTyped. i was assuming that if the types are exported, they might look like "public api" of the plugin, where in fact they are just duplicating cypress types.

also the original test passes
```
addMatchImageSnapshotPlugin(() => null, {}); // $ExpectType void
```

i could also revert to that. i was not sure what exactly was supposed to be tested with the types. specifically as i defined them myself anyway and i can not check against the real cypress ones, as it would otherwise need to depend on it.

what do you think? export as "public api", revert to original test? or the duplication?